modules/oe_theme_helper/src/Plugin/field_group/FieldGroupFormatter/FieldInPageNavigationFormatter.php

```diff
@@ -0,0 +1,188 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_theme_helper\Plugin\field_group\FieldGroupFormatter;
+
+use Drupal\field_group\FieldGroupFormatterBase;
+
+/**
+ * Format a field group to create in-page navigation.
+ *
+ * @FieldGroupFormatter(
+ *   id = "oe_theme_helper_in_page_navigation",
+ *   label = @Translation("Field in-page navigation"),
+ *   description = @Translation("Format field group for ecl-inpage-navigation."),
+ *   supported_contexts = {
+ *     "view"
+ *   }
+ * )
+ */
+class FieldInPageNavigationFormatter extends FieldGroupFormatterBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function process(&$element, $processed_object) {
+    $element += [
+      '#type' => 'field_group_in_page_navigation',
+    ];
+    // Get children elements from the in-page navigation group.
+    $field_elements = $processed_object['#fieldgroups'][$element['#group_name']]->children;
+    $links = [];
+    foreach ($field_elements as $value) {
+      // Check if we should add the field to page contents.
+      if (!is_array($element[$value]) || $this->excludeField($value, $element[$value], $processed_object['#fieldgroups']) === TRUE) {
```

Comment:
Determining emptiness in a render array is known to be problematic in Drupal, the excludeField method tries to do that by trying to handle all possible cases (namely `field` and `group`), but that's not ideal as it doesn't take care of, for example, extra fields.

We had a similar issue on `oe_theme_helper_field_list_pattern` and, at the end, we chose to render the fields in the template and check for emptiness there:

```twig
{% for _item in items %}
  {% set _body = _item.body|render %}
  {% if _body %}
    {% set _items = _items|merge([{
      term: _item.label,
      definition: _body
    }]) %}
  {% endif %}
```

I suggest to do the same here and to get rid of all the "establish if empty" logic.

Reply:
I moved the check to the template, and removed it from the formatter